packages/serverpod/lib/src/server/serverpod.dart

```diff
@@ -374,11 +374,21 @@ class Serverpod {
           config: config,
           experimentalFeatures: experimentalFeatures,
         ) {
-    _initializeServerpod(
-      args,
-      config: config,
-      experimentalFeatures: experimentalFeatures,
-    );
+    try {
+      _initializeServerpod(
+        args,
+        config: config,
+        experimentalFeatures: experimentalFeatures,
+      );
+    } on ExitException catch (e) {
+      if (e.message.isNotEmpty) {
+        stderr.writeln(e.message);
+      }
+      exit(e.exitCode);
+    } catch (e, stackTrace) {
+      _reportException(e, stackTrace, message: 'Error initializing Serverpod');
+      throw ExitException(1, 'Serverpod initialization failed: $e');
```

Comment:
Wouldn't this now get printed twice?

(here and in the parent, which usually does not have any exception handling the main method and thus just crashes).

Probably not a huge deal, but I would currently thing that print + exit is the better choice here then as well. (matching the later `init`)

Also for logging, it might be easier to identify the issue with the original exception than having it wrapped and the type discarded here, no?

Reply:
No because we catch the `ExitException` and print it's message

Reply:
> Wouldn't this now get printed twice?

But that's what I mean. First through `_reportException` and then higher up by catching the `ExitException`.

Reply:
Currently, the error messages look like:

```console
SERVERPOD version: 2.9.1, dart: 3.8.0 (stable) (Wed May 14 09:07:14 2025 -0700) on "macos_arm64", time: 2025-07-07 07:57:41.728921Z
Missing password for "database". Please check your config/passwords.yaml file or the SERVERPOD_PASSWORD_database environment variable.
```

The `ExitException` thrown falls through, and the user is responsible for catching it else it's unhandled in the current code flow

Reply:
```suggestion
      exit(1);
```

Since the error + stack is already printed, and thus would align with the behavior of `Serverpod.start()`